From 8b13d737d3c8add9de39e48955afd407f8f5eb78 Mon Sep 17 00:00:00 2001
From: Brahma Reddy Battula <brahma@apache.org>
Date: Thu, 27 Oct 2016 19:46:41 +0530
Subject: [PATCH 2036/2748] HADOOP-13201. Print the directory paths when
 ViewFs denies the rename operation on internal
 dirs. Contributed by Tianyin Xiu

(cherry picked from commit 0c837db8a874079dd5db83a7eef9c4d2b9d0e9ff)
(cherry picked from commit 616d6345256ec275a08cd55fb3d17fd6146710a0)

 Conflicts:
	hadoop-common-project/hadoop-common/CHANGES.txt

Change-Id: I5f030154efe4a39b5b37f0503b8aa6948cb6d97c
---
 .../java/org/apache/hadoop/fs/viewfs/ViewFs.java   |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFs.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFs.java
index 4610e33..29716d5 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFs.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFs.java
@@ -463,14 +463,16 @@ public void renameInternal(final Path src, final Path dst,
   
     if (resSrc.isInternalDir()) {
       throw new AccessControlException(
-          "Cannot Rename within internal dirs of mount table: it is readOnly");
+          "Cannot Rename within internal dirs of mount table: src=" + src
+              + " is readOnly");
     }
-      
+
     InodeTree.ResolveResult<AbstractFileSystem> resDst = 
                                 fsState.resolve(getUriPath(dst), false);
     if (resDst.isInternalDir()) {
       throw new AccessControlException(
-          "Cannot Rename within internal dirs of mount table: it is readOnly");
+          "Cannot Rename within internal dirs of mount table: dest=" + dst
+              + " is readOnly");
     }
     
     /**
-- 
1.7.9.5

